agera/src/main/java/com/google/android/agera/RepositoryCompilerStates.java

```diff
@@ -469,6 +470,33 @@
     RConfig<TVal> onConcurrentUpdate(@RepositoryConfig int concurrentUpdateConfig);
 
     /**
+     * Specifies a {@code disposer} to handle any intermediate values discarded from the data
+     * processing flow. Intermediate values may be discarded due to the rest of the flow being
+     * skipped by a skip directive or a termination clause ({@code thenSkip()}, {@code orSkip()} and
+     * {@code orEnd()}), or cancelled in compliance with the deactivation or concurrent update
+     * configuration. The disposer may want to check if the values should be disposed of in a proper
+     * way, such as calling {@link Closeable#close()} on all {@link Closeable} values.
+     *
+     * <p>In each occasion where an intermediate value is discarded, the disposer will receive the
+     * value computed by the last completed directive. If the last directive was an attempt that
+     * failed, the disposer will receive the failed {@link Result}.
+     *
+     * <p>The compiled repository will only avoid sending the current repository value (accessible
```

Comment:
Thanks for the clarification, makes sense now. Not sure what I was thinking, re-reading it now and I get it.